src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/SqlColumnEncryptionAzureKeyVaultProvider.cs

```diff
@@ -230,45 +233,36 @@ public override byte[] DecryptColumnEncryptionKey(string masterKeyPath, string e
             int signatureLength = encryptedColumnEncryptionKey.Length - currentIndex - cipherTextLength;
             if (signatureLength != keySizeInBytes)
             {
-                throw new ArgumentException(String.Format(CultureInfo.InvariantCulture, Strings.InvalidSignatureLengthTemplate,
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, Strings.InvalidSignatureLengthTemplate,
                                                             signatureLength,
                                                             keySizeInBytes,
                                                             masterKeyPath),
                                             Constants.AeParamEncryptedCek);
             }
 
             // Get ciphertext
-            byte[] cipherText = new byte[cipherTextLength];
-            Buffer.BlockCopy(encryptedColumnEncryptionKey, currentIndex, cipherText, 0, cipherTextLength);
+            byte[] cipherText = encryptedColumnEncryptionKey.Skip(currentIndex).Take(cipherTextLength).ToArray();
             currentIndex += cipherTextLength;
 
             // Get signature
-            byte[] signature = new byte[signatureLength];
-            Buffer.BlockCopy(encryptedColumnEncryptionKey, currentIndex, signature, 0, signature.Length);
+            byte[] signature = encryptedColumnEncryptionKey.Skip(currentIndex).Take(signatureLength).ToArray();
 
             // Compute the hash to validate the signature
-            byte[] hash;
-            using (SHA256 sha256 = SHA256.Create())
-            {
-                sha256.TransformFinalBlock(encryptedColumnEncryptionKey, 0, encryptedColumnEncryptionKey.Length - signature.Length);
-                hash = sha256.Hash;
-            }
+            byte[] hash = encryptedColumnEncryptionKey.Take(encryptedColumnEncryptionKey.Length - signatureLength).ToArray();
```

Comment:
This is still the preimage at this point and not a hash yet since keyvault does the hashing now. The new library makes this variable name misleading now.